### PR TITLE
Simple test for unit testing the helpers/cache.js

### PR DESCRIPTION
**Pull request by reasonthearchitect**
Another unit test. This one covers the helpers/cache.js. No new dependancies are introduced and only two files to commit.
/test/unit/helpers/cache.js
/test/unit/index.js

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,32 +100,27 @@ pm2 start --name lisk app.js -- -p [port] -a [address] -c [config-path]
...
-```
-
-Launch Lisk (runs on port 4000):
+Launch Lisk (runs on port 4000). This command will set the NODE_ENV=TEST.
+Additionally, it will load the configuration from the 'test' folder rather than the main folder.
+Finally. it will drop the list_test DB and recreate it.
 
 ```
-node app.js
+npm run start-test-server
```
Comment (Isabello) [inline README.md]: There is no associated change in package.json to support this?

**Comment by reasonthearchitect** [comment]
Nice find. Issues with managing to many branches and moving around to much between then... seems like it got pushed in.

In any event.... I noticed that you have accepted the change for the README.md file... as a result, I have updated this branch with that copy.

**Comment by jwkevinc** [comment]
keep up the great work

File: test/unit/helpers/cache.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	it('is not enabled', function () {
+		cache.connect(false, {}, {}, function (err, result) {
+			expect(result.cacheEnabled).to.eq( false );
+			expect(result.client).to.be.null;
+		});
+	});
+
+	it('is enabled redis client callback ready', function () {
+
+		var logger = {
```
Comment (SargeKhan) [inline test/unit/helpers/cache.js]: logger and mockredisclient are duplicated in each test, it'd be nicer if you declare them in the describe scope and use the same object in each test.
Also, update `mockredisclient` variable name to camel case.

File: test/unit/helpers/cache.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		};
+
+		var mockedRedis = sinon.mock(redis);
+		mockedRedis.expects('createClient').once().withArgs(sinon.match.any).returns(mockredisclient);
+		cache.connect(true, {}, logger, function (err, result) {
+			expect(result.cacheEnabled).to.eq( true );
+			expect(result.client).to.be.not.null;
+
+			mockedRedis.restore();
+			mockedRedis.verify();
```
Comment (SargeKhan) [inline test/unit/helpers/cache.js]: Sinon mock's `verify` function internally calls `restore` function. So, here, `mockedRedis.restore()` is redundant.

File: test/unit/helpers/cache.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+		});
+	});
+
+	it('is enabled redis client callback error', function () {
+
+		var logger = {
+			'error': function (message, err) {
+				expect(message).to.eq('Redis:');
+				expect(err + '').to.eq('Error: Cache Test Callback Error');
```
Comment (SargeKhan) [inline test/unit/helpers/cache.js]: You can do deep equality comparison with `.eql` function. So, please don't perform implicit conversion of Error to string value.

File: test/unit/helpers/cache.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+			}
+		};
+
+		var config = {password: null};
+		expect(config).to.have.property('password');
+		var mockedRedis = sinon.mock(redis);
+		mockedRedis.expects('createClient').once().withArgs(sinon.match.any).returns(mockredisclient);
+		cache.connect(true, config, logger, function (err, result) {
+			expect(result.cacheEnabled).to.eq( true );
+			expect(result.client).to.be.not.null;
```
Comment (SargeKhan) [inline test/unit/helpers/cache.js]: You should update the assertion by expecting `result.client` to eql `mockredisclient`.

File: test/unit/helpers/cache.js (hunk 2 above)
Comment (reasonthearchitect) [reply test/unit/helpers/cache.js]: No problem. Moved the logger up to the describe scope.

File: test/unit/helpers/cache.js (hunk 3 above)
Comment (reasonthearchitect) [reply test/unit/helpers/cache.js]: Thanks for pointing this out... I was not aware of that!

Updated the tests to create in the beforeEach() method and verify in the afterEach() method.

File: test/unit/helpers/cache.js (hunk 4 above)
Comment (reasonthearchitect) [reply test/unit/helpers/cache.js]: Added. Made declare scope var ERROR.

File: test/unit/helpers/cache.js (hunk 5 above)
Comment (reasonthearchitect) [reply test/unit/helpers/cache.js]: Nice... did it in both places that this occurs.

**Comment by reasonthearchitect** [comment]
SargeKhan. Thanks for the great review... you guys are ninjas... :)  Think the code looks cleaner now as a result.

One note of interest is the suggestion to move the mockRedisClient into the declare scope. I did one called mockRedisClientReady as it is leveraged by 2 tests... however, I left the mockRedisClientError in the method scope as it is only used once. Note that we cannot collapse this into one object as the unit under test calls both [ client.on('success', cb) && client.on('error', cb) ]. As a result, we must maintain these clients as two separate variables.

**Comment by reasonthearchitect** [comment]
Thanks for the review guys!

**Pull request merged**